app/src/main/java/io/appium/settings/RecorderThread.java

```diff
@@ -0,0 +1,419 @@
+package io.appium.settings;
+
+import android.hardware.display.DisplayManager;
+import android.hardware.display.VirtualDisplay;
+import android.media.AudioAttributes;
+import android.media.AudioFormat;
+import android.media.AudioPlaybackCaptureConfiguration;
+import android.media.AudioRecord;
+import android.media.MediaCodec;
+import android.media.MediaCodecInfo;
+import android.media.MediaFormat;
+import android.media.MediaMuxer;
+import android.media.projection.MediaProjection;
+import android.os.Build;
+import android.os.Handler;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.Surface;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import androidx.annotation.RequiresApi;
+
+public class RecorderThread implements Runnable {
+
+    private static final String TAG = "RecorderThread";
+
+    private MediaProjection mediaProjection;
+    private VirtualDisplay virtualDisplay;
```

Comment:
do we really need to share so many variables across the instance?

Reply:
i may try to reduce it via making them local vars but there is not so much of that type convertible-vars in it currently if i am not wrong

Reply:
I personally find it much more convenient to work with something that looks like

```
public X someMethod() { return new X;}

public void someOtherMethod () {
  X x = someMethod();
}
```

rather than 

```

private x;

private void someMethod() {
   this.x = ...;
}

public void someOtherMethod() {
  someMethod();
  // do something this this.x
}
```

Basically the second one is hard to parse because it assumes an implicit dependency of someOtherMethod on someMethod. If someMethod is not called before someOtherMethod then we'd get an unexpected NPE because of uninitialised `x`. 

The first option makes it more obvious that we need to call someMethod in order to have the instance of X. This becomes critical if there are several variables similar to `x`. The rule of thumb here is to limit the scope of a variable to the necessary minimum and don't expand it unless it is required. Almost the same is also applicable to variables privacy level and to sizes of local variable visibility namespaces.